route.go

```diff
@@ -43,6 +43,8 @@ func (r *Route) Match(req *http.Request, match *RouteMatch) bool {
 		return false
 	}
 
+	//reset MatchErr
```

Comment:
Tell a story / why - it's obvious that we reset it :)

Instead, say: "Set MatchErr to nil to prevent subsequent matching subrouters from failing to run middleware. If not reset, the middleware would see a non-nil MatchErr and be skipped, even when there was a matching route"